gnomad_qc/v3/analysis/vary_subpop_params.py

```diff
@@ -0,0 +1,215 @@
+            hl.literal(
+                merged,
+                "array<struct{min_prob: float, max_proportion_mislabeled_training_samples: float, percent_correct: float, TP: int, FP: int, FN: int, error_rate: float}>",
+            )
+        )
+
+        # Calculate precision and recall
+        # Precision: TP/(TP+FP)
+        # Recall: TP/(TP+FN)
+        results_ht = results_ht.annotate(
+            precision=results_ht.TP / (results_ht.TP + results_ht.FP),
+            recall=results_ht.TP / (results_ht.TP + results_ht.FN),
+        )
+
+        # Write results
+        file_path = get_checkpoint_path(
+            "subpop_analysis/parameter_testing_{}{}{}".format(
+                pop,
+                "_with_unreleasable_samples" if include_unreleasable_samples else "",
+                ".high_quality" if high_quality else "",
+            )
+        )
+
+        results_ht = results_ht.checkpoint(file_path, overwrite=args.overwrite)
+
+    finally:
+        logger.info("Copying hail log to logging bucket...")
+        hl.copy_log(get_logging_path("subpop_parameter_testing"))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="This script generates a Table with precision and recall results to test the performance of subpop assignment"
+    )
+    parser.add_argument(
+        "--slack-channel",
+        help="Slack channel to post results and notifications to",
+    )
+    parser.add_argument(
+        "--overwrite", help="Overwrites existing files", action="store_true"
+    )
+    parser.add_argument(
+        "--pop",
+        help="Population to test (need to first run subpop_analysis.py with this `pop`",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--min-prob-list",
+        help="List of minimum RF probabilities to test for subpop assignment",
+        type=float,
+        nargs="+",
+        default=[
+            0.10,
+            0.20,
+            0.30,
+            0.40,
+            0.50,
+            0.60,
+            0.70,
+            0.75,
+            0.80,
+            0.85,
+            0.90,
+            0.95,
+        ],
+    )
+    parser.add_argument(
+        "--max-prop-mislabeled-list",
+        help="List of maximum number of training samples that can be mislabelled to test for subpop assignment",
+        type=float,
+        nargs="+",
+        default=[1],
+    )
+    parser.add_argument(
+        "--pcs",
+        help="List of PCs to use in the subpop RF assignment",
+        type=int,
+        nargs="+",
+        default=list(range(1, 20)),
+    )
+    parser.add_argument(
+        "--withhold-prop",
+        help="Proportion of training pop samples to withhold from training, all samples will be kept if this flag is not used. NOTE: 20 percent of the training samples are used for evaluation, withholding samples will put aside a certain proportion of samples before splitting the remainder into training and evaluation",
```

Comment:
Thanks for adding this note, maybe also add that this is very important to do if `max-prop-mislabeled-list` is supplied as this is the only way to give a reasonable/helpful PR curve if iterations are performed